packages/libs/client-utils/src/nodejs/marmalade/deployment/deploy.ts

```diff
@@ -0,0 +1,199 @@
+  defaultRemoteConfig,
+  defaultRepositoryConfig,
+} from './defaults';
+import { deployMarmaladeNamespaces } from './namespace';
+
+interface IDeployMarmaladeInput {
+  sender?: IAccount;
+  chainIds?: ChainId[];
+  localConfig?: ILocalConfig;
+  remoteConfig?: IRemoteConfig;
+  repositoryConfig?: IRepositoryConfig;
+  namespaceConfig?: INamespaceConfig[];
+  namespaceDeployOrder?: string[];
+  deploymentArguments?: Record<string, any>;
+  clientConfig?: IClientConfig;
+  deleteFilesAfterDeployment?: boolean;
+}
+
+/**
+ * Gets marmalade template, contracts and namespaces from the remote repository and deploys them to the chain.
+ * @alpha
+ * @param sender - The sender account for deploy marmalade namespaces
+ * @param chainIds - The chain ids where the marmalade will be deployed
+ * @param localConfig - The local configuration for the marmalade deployment: where the template, code files and namespace files will be stored
+ * @param remoteConfig - The remote configuration for the marmalade deployment: the path to the marmalade templates and code files, the file extensions and the exclude folders
+ * @param repositoryConfig - The repository configuration for the marmalade deployment: repository related data and the github token
+ * @param namespaceConfig - The namespace configuration for the marmalade deployment: the namespace files and the namespaces to be deployed
+ * @param namespaceDeployOrder - The order in which the namespaces will be deployed
+ * @param deploymentArguments - The deployment arguments for the marmalade deployment: these will replace the holes in the marmalade templates
+ * @param clientConfig - The client configuration for the marmalade deployment: the host, the network id and the sender key
+ * @param deleteFilesAfterDeployment - If true, the local files will be deleted after the deployment
+ */
+export const deployMarmalade = async ({
+  sender = defaultAccount,
+  chainIds = [defaultChainId],
+  localConfig = defaultLocalConfig,
+  remoteConfig = defaultRemoteConfig,
+  repositoryConfig = defaultRepositoryConfig,
+  namespaceConfig = defaultNamespaceConfig,
+  namespaceDeployOrder = defaultNamespaceDeployOrder,
+  deploymentArguments = defaultArguments,
+  clientConfig = defaultClientConfig,
+  deleteFilesAfterDeployment = false,
+}: IDeployMarmaladeInput) => {
+  try {
+    // Merge the default arguments with the provided arguments
+    deploymentArguments = { ...defaultArguments, ...deploymentArguments };
+    console.log(deploymentArguments);
+
+    console.log('Preparing directories...');
+    handleDirectorySetup(
+      localConfig.templatePath,
+      localConfig.codeFilesPath,
+      localConfig.namespacePath,
+    );
+
+    console.log(
+      'Downloading marmalade templates and namespace definition files...',
+    );
+
+    await Promise.all([
+      getMarmaladeTemplates({
+        repositoryConfig,
+        remoteConfig,
+        localConfig,
+        flatFolder: true,
+      }),
```

Comment:
What do these functions do? Their results aren't used, or they return nothing resulting in a side-effect. 

I suppose they write to disk. I'm not sure how I feel about "unused" function calls


_Idem for other calls below_